### ISSUE TITLE
broken session key deletion in 5.2.0

### ISSUE THREAD

**Issue by igorkasyanchuk** [redis-store/redis-actionpack#29]
An issue with pretender gem (and I support with other things which are using "session") when session_store is in redis.

Code below
```
  def logout
    logger.info 'Stop Impersonating...'.red
    stop_impersonating_user
   # stop_impersonating_user - from gem pretender
   # it basically deleting session key
   # https://github.com/ankane/pretender/blob/fd89162912dcb5584c703948beb3dd7f83e958cf/lib/pretender.rb#L64
   # request.session.delete("impersonated_user_id")
    redirect_to '/'
  end
```

and even in after_action I could see that key is deleted, but on next page load key is present.

If I downgrade to version 5.1.0 all works.

**Comment by igorkasyanchuk** [comment]
actually looks like issue in the latest redis-rack.

Because with:
gem 'redis-rack', '2.0.6'
all works.

So this is a combination of code which works for me:
gem 'redis-actionpack', '5.1.0'
gem 'redis-rack', '2.0.6'

**Comment by igorkasyanchuk** [comment]
git diff to show how I fixed issue locally:
```diff
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -678,21 +678,14 @@ GEM
     rb-inotify (0.10.1)
       ffi (~> 1.0)
     redis (4.1.3)
-    redis-actionpack (5.2.0)
-      actionpack (>= 5, < 7)
-      redis-rack (>= 2.1.0, < 3)
+    redis-actionpack (5.1.0)
+      actionpack (>= 4.0, < 7)
+      redis-rack (>= 1, < 3)
       redis-store (>= 1.1.0, < 2)
-    redis-activesupport (5.2.0)
-      activesupport (>= 3, < 7)
-      redis-store (>= 1.3, < 2)
     redis-namespace (1.7.0)
       redis (>= 3.0.4)
-    redis-rack (2.1.1)
-      rack (>= 2.0.8, < 3)
-      redis-store (>= 1.2, < 2)
-    redis-rails (5.0.2)
-      redis-actionpack (>= 5.0, < 6)
-      redis-activesupport (>= 5.0, < 6)
+    redis-rack (2.0.6)
+      rack (>= 1.5, < 3)
       redis-store (>= 1.2, < 2)
     redis-store (1.8.1)
       redis (>= 4, < 5)
```

**Comment by tubbo** [comment]
were you on redis-rack 2.1.1? the code in v2.1.0 was reverted because it was causing problems like this.

**Comment by tubbo** [comment]
hey sorry about that my version numbers were all mixed up...I just released v2.1.2 which should fix the problem you're having. let me know if it doesn't work.

**Comment by igorkasyanchuk** [comment]
Yeap, works now. Thanks
